Userland/Applications/PixelPaint/Tools/MoveTool.cpp

```diff
@@ -188,17 +191,56 @@ void MoveTool::on_keyup(GUI::KeyEvent& event)
 {
-    if (layer != m_layer_being_moved.ptr() || !m_scaling)
-        return;
-
+    VERIFY(m_editor);
     GUI::Painter painter(*m_editor);
     painter.add_clip_rect(event.rect());
-    auto rect_in_editor = m_editor->content_to_frame_rect(m_new_layer_rect).to_rounded<int>();
-    painter.draw_rect(rect_in_editor, Color::Black);
-    if (!m_cached_preview_bitmap.is_null() || !update_cached_preview_bitmap(layer).is_error()) {
+    auto content_rect = m_scaling ? m_new_layer_rect : m_editor->active_layer()->relative_rect();
+    auto rect_in_editor = m_editor->content_to_frame_rect(content_rect).to_rounded<int>();
+    if (m_scaling && (!m_cached_preview_bitmap.is_null() || !update_cached_preview_bitmap(layer).is_error())) {
+        Gfx::PainterStateSaver saver(painter);
         painter.add_clip_rect(m_editor->content_rect());
         painter.draw_scaled_bitmap(rect_in_editor, *m_cached_preview_bitmap, m_cached_preview_bitmap->rect(), 1.0f, Gfx::Painter::ScalingMode::BilinearBlend);
     }
+    painter.draw_rect_with_thickness(rect_in_editor, Color::Black, 3);
+    painter.draw_rect_with_thickness(rect_in_editor, Color::White, 1);
+    auto size = resize_anchor_size(rect_in_editor);
+    if (size < resize_anchor_min_size)
+        return;
+
+    auto resize_anchors = resize_anchor_rects(rect_in_editor, size);
+    for (auto const& resize_anchor_rect : resize_anchors) {
+        painter.draw_rect_with_thickness(resize_anchor_rect, Color::Black, 3);
+        painter.draw_rect_with_thickness(resize_anchor_rect, Color::White, 1);
+    }
+}
+
+Gfx::IntRect MoveTool::resize_anchor_rect_from_position(Gfx::IntPoint position, int size)
+{
+    auto resize_anchor_rect_top_left = position.translated(-size / 2);
+    return Gfx::IntRect(resize_anchor_rect_top_left, Gfx::IntSize(size, size));
+}
+
+int MoveTool::resize_anchor_size(Gfx::IntRect layer_rect_in_frame_coordinates)
+{
+    auto shortest_side = min(layer_rect_in_frame_coordinates.width(), layer_rect_in_frame_coordinates.height());
+    if (shortest_side <= 1)
+        return 1;
+    int x = ceilf(shortest_side / 3.0f);
+    return min(resize_anchor_max_size, x);
+}
+
+Vector<Gfx::IntRect, 4> MoveTool::resize_anchor_rects(Gfx::IntRect layer_rect_in_frame_coordinates, int resize_anchor_size)
+{
+    Vector<Gfx::IntRect, 4> rects;
+    auto add_resize_anchor_rect = [&rects, resize_anchor_size](Gfx::IntPoint position) {
+        auto resize_anchor_rect = resize_anchor_rect_from_position(position, resize_anchor_size);
+        rects.append(resize_anchor_rect);
+    };
+    add_resize_anchor_rect(layer_rect_in_frame_coordinates.top_left());
+    add_resize_anchor_rect(layer_rect_in_frame_coordinates.top_right().translated(1, 0));
+    add_resize_anchor_rect(layer_rect_in_frame_coordinates.bottom_left().translated(0, 1));
+    add_resize_anchor_rect(layer_rect_in_frame_coordinates.bottom_right().translated(1));
+    return rects;
```

Comment:
I think this might be a little simpler just to directly return an Array literal rather than append to a vector:
```suggestion
Array<Gfx::IntRect, 4> MoveTool::resize_anchor_rects(Gfx::IntRect layer_rect_in_frame_coordinates, int resize_anchor_size)
{
    auto resize_anchor_rect = [&rects, resize_anchor_size](Gfx::IntPoint position) {
        return resize_anchor_rect_from_position(position, resize_anchor_size);
    };
    return Array {
        resize_anchor_rect(layer_rect_in_frame_coordinates.top_left()),
        resize_anchor_rect(layer_rect_in_frame_coordinates.top_right().translated(1, 0)),
        resize_anchor_rect(layer_rect_in_frame_coordinates.bottom_left().translated(0, 1)),
        resize_anchor_rect(layer_rect_in_frame_coordinates.bottom_right().translated(1))
    };
```

Reply:
Done. Thanks! :^)